src/edu/washington/escience/myria/coordinator/catalog/MasterCatalog.java

```diff
@@ -1805,7 +1808,8 @@ private void deleteRelationIfExists(@Nonnull final SQLiteConnection sqliteConnec
       @Nonnull final RelationKey relation) throws CatalogException {
     try {
       SQLiteStatement statement =
-          sqliteConnection.prepare("DELETE FROM relations WHERE user_name=? AND program_name=? AND relation_name=?;");
+          sqliteConnection
+              .prepare("DELETE FROM relations WHERE user_name=? AND program_name=? AND relation_name=? AND tombstone=1;");
```

Comment:
On MasterCatalog, the ```relation_schema``` table is declared with the following constraint.... ```FOREIGN KEY (user_name,program_name,relation_name) REFERENCES relations ON DELETE CASCADE);```

So if ``relations``` deletes a row, it should be deleted from ```relation_schema``` and  ```stored_relations``` as well I believe